code/game/machinery/abominationmachine.dm

```diff
@@ -0,0 +1,179 @@
+	
+	//Reset the machine
+	has_creature = FALSE
+	n_arms_eaten = 0
+	n_legs_eaten = 0
+	n_organs_eaten = 0
+	n_ublood_eaten = 0
+	creature_name = ""
+	reagents.clear_reagents()
+	update_icon()
+	
+	return creature
+
+/obj/machinery/abomachine/obj_destruction()	//what is the difference?
+	release_creature(TRUE)
+	..()
+
+/obj/machinery/abomachine/Initialize()
+	. = ..()
+	create_reagents(300, OPENCONTAINER)
+	update_icon()
+	
+/obj/machinery/abomachine/update_icon()
+	icon_state = has_creature ? "pod_g" : "pod_0"
+
+/obj/machinery/abomachine/examine(mob/user)
+	. = ..()
+	. += {"A digital display on it reads "[seconds_remaining()]"."}
+
+/obj/machinery/abomachine/proc/feed_part(obj/item/part)
+	if (istype(part, obj/item/bodypart))//can't feed non body parts		
+		if (!has_creature)//no creature inside
+			return FALSE
+		var/obj/item/bodypart/limb = part
+		if (limb.status != BODYPART_ORGANIC)
+			return FALSE	//cannot feed robotic body parts either
+		
+		//arms 
+		if (istype(limb, obj/item/bodypart/r_arm) || istype(limb, obj/item/bodypart/l_arm))		
+			//if (proc(acceptchance)) 		RNG bad, use hardcap instead
+				n_arms_eaten = n_arms_eaten + 1
+		//legs 
+		if (istype(limb, obj/item/bodypart/r_leg) || istype(limb, obj/item/bodypart/l_leg))		
+				n_legs_eaten = n_legs_eaten + 1
+		qdel(limb)
+		return TRUE	//we done it
+	
+	if (istype(part, obj/item/organ))
+		var/obj/item/organ/thing = part
+		if (thing.status != ORGAN_ORGANIC)
+			return FALSE	//cannot feed robotic body parts either
+		
+		if (istype(thing, obj/item/organ/heart))	//have a heart
+			if (!has_creature)				
+				has_creature = TRUE
+				update_icon()
+			else
+				n_organs_eaten = n_organs_eaten + 1
+			qdel(thing)
+			return TRUE	
+			
+		if (!has_creature)//no creature inside
+			return FALSE
+		
+		if (istype(thing, obj/item/organ/brain))
+		
+			to_chat(user, "<span class='notice'>You offer [src] to [SM]...</span>")	
+			var/obj/item/organ/brain/brian = thing
+			
+			if(brian.brainmob && brian.brainmob.mind) //enslave the current mind into the creature
+			
+				var/mob/creature = release_creature(FALSE)
+				brainmob.mind.transfer_to(creature)
+				creature.key = brian.brainmob.mind.key
+				creature.mind.enslave_mind_to_creator(user)
+				creature.sentience_act()
+				to_chat(creature, "<span class='warning'>All at once it makes sense: you know what you are and who you are! Self awareness is yours!</span>")
+				to_chat(creature, "<span class='userdanger'>You are grateful to be self aware and owe [user.real_name] a great debt. Serve [user.real_name], and assist [user.p_them()] in completing [user.p_their()] goals at any cost.</span>")
+				creature.copy_languages(user)	
+			
+			else //create a ghostpoll
+				var/list/candidates = pollCandidatesForMob("Do you want to play as [SM.name]?", ROLE_SENTIENCE, null, ROLE_SENTIENCE, 50, SM, POLL_IGNORE_SENTIENCE_POTION)
+				if(LAZYLEN(candidates))
+				
+					var/mob/creature = release_creature(FALSE)
```

Comment:
A lot of this is the exact same as above. You should write these lines only once and have them run in every condition you have already redundantly written them